components/utils/useSessionGroups.ts

```diff
@@ -63,56 +63,48 @@ export function useSessionGroups(sessions: Session[]): SessionGroups {
+    currentSessionIndex = allSessionGroups.length
+  // Once we are past the first session group we start showing times
+  } else if (isAfter(dateTimeProvider.now().Value, allSessionGroups[0].timeEnd)) {
+    currentSessionIndex = allSessionGroups.findIndex(g => isWithinInterval(dateTimeProvider.now().Value, {
+      start: g.timeStart,
+      end: g.timeEnd,
+    }));
+  }
+  // If we aren't currently in a session, calculate the session that just finished
+  if (currentSessionIndex === -1) {
+    const finishedSessionGroups = allSessionGroups.filter(g => isAfter(dateTimeProvider.now().Value, g.timeEnd));
+    previousSessionIndex = finishedSessionGroups.length - 1;
+  }
 
-    if (isCurrentSession) {
-      pastSessionsIndex = i - 2 >= 0 ? i - 1 : -1
-      previousSessionIndex = i - 1 >= 0 ? i - 1 : -1
-      currentSessionIndex = i
-      nextSessionIndex = i + 1 < allSessionGroups.length ? i + 1 : -1
-      futureSessionIndex = i + 2 < allSessionGroups.length ? i + 2 : -1
-      break
-    } else if (isNextSession) {
-      pastSessionsIndex = i - 2 >= 0 ? i - 1 : -1
-      previousSessionIndex = i - 1 >= 0 ? i - 1 : -1
-      currentSessionIndex = -1
-      nextSessionIndex = i
-      futureSessionIndex = i + 1 < allSessionGroups.length ? i + 1 : -1
-      break
-    } else if (i === allSessionGroups.length - 1 && isAfterSession) {
-      // after the last talk
-      pastSessionsIndex = i - 1
-      previousSessionIndex = i
-      break
-    }
+  // Calculate all of the indexes based on the ones we just calculated
+  previousSessionIndex = currentSessionIndex === -1 ? previousSessionIndex : currentSessionIndex - 1;
+  const pastSessionsIndex = previousSessionIndex - 1;
+  let nextSessionIndex = currentSessionIndex === -1 ? previousSessionIndex + 1 : currentSessionIndex + 1;
+  let futureSessionIndex = nextSessionIndex + 1;
+  // Protect the indexes from growing past the bounds of the number of session groups
+  if (currentSessionIndex >= allSessionGroups.length) {
+    currentSessionIndex = -1;
+  }
+  if (nextSessionIndex >= allSessionGroups.length) {
+    nextSessionIndex = -1;
+  }
+  if (futureSessionIndex >= allSessionGroups.length) {
+    futureSessionIndex = -1;
   }
 
   const calculatedSessionGroups = React.useMemo<Omit<SessionGroups, 'allSessionGroups'>>(
     () => ({
-      ...(pastSessionsIndex !== -1 ? { pastSessionGroups: allSessionGroups.slice(0, pastSessionsIndex) } : {}),
-      ...(previousSessionIndex !== -1 ? { previousSessionGroup: allSessionGroups[previousSessionIndex] } : {}),
-      ...(currentSessionIndex !== -1 ? { currentSessionGroup: allSessionGroups[currentSessionIndex] } : {}),
-      ...(nextSessionIndex !== -1 ? { nextSessionGroup: allSessionGroups[nextSessionIndex] } : {}),
-      ...(futureSessionIndex !== -1 ? { futureSessionGroups: allSessionGroups.slice(futureSessionIndex) } : {}),
+      ...(pastSessionsIndex >= 0 ? { pastSessionGroups: allSessionGroups.slice(0, pastSessionsIndex + 1) } : {}),
```

Comment:
The +1 at the end here was a bug, the rest of the changes in this file are optional, but I think it's a bit clearer (still a bit hard to read though, array indexes are confusing lol)